### PR TITLE
List add&remove

### PR DESCRIPTION
**Pull request by Bupilipili**
### In this branch

-Removed all hardcoded items from the tasks array.
 -Created a new JavaScript file for the new functionality.
 -Implemented a function for adding a new task (add a new element to the array).
 -Implemented a function for deleting a task (remove an element from the array).
 -Implemented a function for editing task descriptions.
-Added every change to the local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,32 +1,76 @@
...
     const items = document.createElement('li');
     items.className = 'list-item';
-    items.innerHTML = `<input class="check-box" type="checkbox">${component.description}`;
+    items.dataset.index = component.index;
+    items.innerHTML = `
+    <div class="listboxflex">
+        <input class="check-box" type="checkbox">
+        <input class="description" value="${component.description}" readonly>
+        <a class="delete-icon"><img class="delete-img" src="${bin}" alt="bin"></a>
+    </div>
```
Comment (OyePriscilla) [inline src/index.js]: - [x] I discovered that this image is broken, I was able to access the functionality through the `alt` attribute. Kindly fix it. 👍 

Untitled

File: src/index.js [hunk 2]
```diff
@@ -1,32 +1,76 @@
 /* eslint-disable no-unused-vars */
+/* eslint-disable prefer-destructuring */
+
```
Comment (OyePriscilla) [inline src/index.js]: - [x] I think it's not a good practise to disable linters, kindly resolve the issues that these linters pointed out. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Bupilipili 👏 👏 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented todo's add, edit and delete functionalities ✔️ 
App worked as expected ✔️ 

## Required Changes ♻️

- [x] Kindly fix the twitter and linkedln broken links in the Author section of the README file, this will make your repo to be more engaging. 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (Bupilipili) [reply src/index.js]: the image is visible

File: src/index.js [hunk 3]
```diff
@@ -1,32 +1,75 @@
 /* eslint-disable no-unused-vars */
+
```
Comment (cynthiainga) [inline src/index.js]: - [ ] As mentioned by the previous reviewer kindly avoid disabling linters, fixing linters help us to write a clean, clear and strong code base, I notice that you disable linters in your server.js file as well. Kindly fix that 🙏🏾 🙏🏾

File: src/index.js [hunk 4]
```diff
@@ -1,32 +1,75 @@
...
   components.forEach((component) => {
     const items = document.createElement('li');
     items.className = 'list-item';
-    items.innerHTML = `<input class="check-box" type="checkbox">${component.description}`;
+    items.dataset.index = component.index;
+    items.innerHTML = `
+    <div class="listboxflex">
+        <input class="check-box" type="checkbox">
+        <input class="description" value="${component.description}" readonly>
+        <a class="delete-icon"><img class="delete-img" src="${bin}" alt="bin"></a>
```
Comment (cynthiainga) [inline src/index.js]: - [ ] As mentioned by the previous reviewer this image is broken, please kindly fix that 🙏🏾 

Screenshot 2023-07-15 at 00 47 15

- [x] Also the index of elements in the array should start with 1 and not 0, (i.e. if a 4th task is added to the list, the index of that task should equal to 4 and not 3). Is part of the project requirements the first reviewer forgot to mention that, please kindly fix that 🙏🏾 

Screenshot 2023-07-15 at 01 09 04

Screenshot 2023-07-15 at 00 45 30

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Bupilipili, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [ ] No linters errors ✔️ 
- [ ] Can add a task to the list ✔️ 
- [ ] Can edit a task ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kamba56 — APPROVED** [review]
Hi @Bupilipili, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] Great job adding the add functionality to your to do list!👏👏👏
- [x] Great job using modules to add funtionalities!👏👏👏
- [x] !👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**